### ISSUE TITLE
Dozens of "File doesn't exist" warnings prevent closing window (even with Update Silently enabled) and shown incorrectly with Network Share

### ISSUE THREAD

**Issue by PowerAccess** [notepad-plus-plus/notepad-plus-plus#1116]
## Several Issues with "File doesn't exist" Showing When Shouldn't
1. Dozens of "File doesn't exist" warnings often prevent me from closing Notepad++ window after deleting or moving a folder. Those warnings shouldn't be shown when closing the application.
2. "File doesn't exist" most often get shown when inappropriate or inaccurate, such as for files on a network share, mapped drive, removable drive, or temp files opened from a zip archive, even when the original file does still exist or becomes available again, the prompts still get shown.
3. "File doesn't exist" shouldn't be shown when "Update Silently" is enabled (though another option could be added if felt needed for these). 

If absolutely necessary, such a warning could just be deferred until if/when the user attempts to save the file to disk, or a new option to allow users to show this warning could be added if needed, but most of the complaints and questions about File Status Detection are related to avoiding this annoying warning.

Please note that disabling File-status Detection is not a viable workaround as that is needed for the primary use case of preventing unintended overwriting of files modified on disk.
## Details

As seen in the below screenshot, the "Keep non existing file" warning prompting for "The file {FilePath} doesn't exist anymore. Keep this file in editor?" appears when the file is no longer found on disk (whether lost connection to network share or removed portable drive, or manually moved deleted before going to close Notepad++), sometimes even repeated 2x for such an event, even when "Update silently" preference is enabled. 
## Screenshot

![unnecessary non existing file warnings despite update silently](https://cloud.githubusercontent.com/assets/4053734/11049804/d7a38c12-86f5-11e5-857e-6267904a96c9.png)
## Test Cases & Common Causes

For files on a network share (whether NAS, VPN, FTP, WebDAV, etc.) or even on a thumb drive or portable drive, any time the connection is lost or drive is removed, then this annoying prompt shows up and then another prompt is shown again once the file becomes available again.  T

Also, these warnings occur when opening a temp file such as from a zip archive, once it's no longer available.

Even for local files, this often occurs when I am done with a temp folder or ready to move it after previewing its contents, and then go to close Notepad, and get prompted for nearly every single open tab before I can safely close the app.

Also, these 1-2 warnings x # of files open (as they are often from the same drive/folder), despite no actual change in the modified timestamp and even when "Silently update" is enabled.  For those who use Notepad most often just for viewing files, instead of editing them, this is especially an unnecessary annoyance. It is even more annoying that 
## Suggested Fix:

Completely disabling File Status Auto-Detection is _not_ a viable option, because then it is far too easy too unknowingly overwrite other changes (from GIT, Dropbox, another app, etc.) to a file, or at least fail to view the latest version of it.

Instead, please add a separate option for "Ignore file exists" "Warn/Prompt/Ignore if file no longer on disk" to Preferences > MISC > File Status Auto-Detection.
## Thousands of Others with Same Complaint

Many others have this same concern, as can be seen in the [832 posts on the SourceForge discussion lists] and [32,000 Google Search results](https://www.google.com/search?q=Notepad%2B%2B+ignore+file+not+available&ie=utf-8&oe=utf-8#q=Notepad%2B%2B+non-existing|%22does+not+exist%22+file|location) about how to avoid this warning, only counting those specifically about the unwanted "Does not exist" warning, as opposed to the useful file modified warning/auto-reloading.

This warning is by far the most annoying thing about Notepad++, occurring and the main reason I've been considering using an alternative, as there is no way to avoid often 10 warnings every time I use Notepad++ while still safely avoiding overwriting actually modified files (with file modified detection), so I (and seemingly many others) would greatly appreciate if you could fix this, ideally by adding a new preference like the one suggested.

**Comment by xeonx1** [comment]
Is this something that can please be addressed soon or an ETA or response provided?

**Comment by xeonx1** [comment]
@donho @milipili 

1. Dozens of "File doesn't exist" warnings often prevent me from closing Notepad++ window after deleting or moving a folder. Those warnings shouldn't be shown when closing the application.

2. "File doesn't exist" most often get shown when inappropriate or inaccurate, such as for files on a network share, mapped drive, removable drive, or temp files opened from a zip archive, even when the original file does still exist or becomes available again, the prompts still get shown.

3. "File doesn't exist" shouldn't be shown when "Update Silently" is enabled (though another option could be added if felt needed for these). 

If absolutely necessary, such a warning could just be deferred until if/when the user attempts to save the file to disk, or a new option to allow users to show this warning could be added if needed, but most of the complaints and questions about File Status Detection are related to avoiding this annoying warning.

Please note that disabling File-status Detection is not a viable workaround as that is needed for the primary use case of preventing unintended overwriting of files modified on disk.

**Comment by xeonx1** [comment]
@donho @milipili @ClaudiaFrank 
Is this someone able to fix this?
Most often this prevents closing Notepad++, as when going to exit the app, it then prompts for each file (most of which were temporary from WinSCP, opened from zip archive, folder that was moved, before lost internet connection, etc.). I have often had to confirm dozens of warnings after clicking X to close the app, warnings that have no purpose when closing the app, and which shouldn't be shown when Update Silently is enabled.

**Comment by timdonovanuk** [comment]
Please fix! This makes working with files on network shares a nightmare. Worse it freezes notepad++ for minutes.

**Comment by schittli** [comment]
Hello
It would be great **if we could configure how Notepad++ should answer the question by default.** 
**Of course I would choose "Yes, keep the File in Editor".** Always, without having to think twice. :-)

This Notepad++ behavior is really very annoying, especially if users often switch between VPN networks or between LAN and WLAN.

Thank you very much, best wishes,
Thomas

**Comment by mijioij** [comment]
I don't know why this was closed. When I enabled the setting (Settings -> MISC. -> File Status Auto-Detection -> Enable for all opened files), for files in mapped drives when the mapped connection closes, I still get a popup saying "The file doesn't exist anymore. Keep this file in editor?".  

I want a setting to always keep a file in editor even if it doesn't exist anymore so the "...Keep this file in editor?" message doesn't show up.
